### ISSUE TITLE
Library page - Speed up test execution

### ISSUE THREAD

**Issue by tbrlpld** [MozillaFoundation/foundation.mozilla.org#10391]
## Description

Currently, the ~30 tests for the library page take me locally about 80-120 seconds to run. This is way too slow. 
We should speed this up to facilitate faster development cycles. 

You can probably move setup code to `setUpTestData` and switch from using the test client to testing the getter method for the detail pages directly.

[Epic](https://github.com/MozillaFoundation/foundation.mozilla.org/issues/10339) | [Designs](https://www.figma.com/file/dPGD3acNbT75dvEGzNaGUb/Responsible-Computing?t=zoMb9zxg25R0T6yt-6)

## Acceptance criteria

- [ ] As a developer, I can confirm that running the tests for the library page is faster than the baseline execution time I get with the target branch. 

:warning: Depends on #10392, as that ticket will be making changes to the functions that we will be using when updating the tests.

**Comment by tbrlpld** [comment]
@danielfmiranda Let me know if you have any questions here.

**Comment by danielfmiranda** [comment]
Thanks @tbrlpld!

Interestingly, my usual time to run all the test is about 230 seconds instead of  80-120.

After updating some of these to use the getter method like you mentioned above, I was able to bring my total runtime down around ~40%. I have been tracking test runtimes in this [google sheets doc](https://docs.google.com/spreadsheets/d/1JKB4CaOlfRldns8O5YfF5sFSTxjAa9cOIRlQmcoQ6-s/edit#gid=0). 

However, I have been experiencing an issue when trying to work on the test `test_filter_multiple_profiles` and I cant seem to find the cause of the issue. I was wondering if you might be able to help.

To start, I am trying to return a queryset of all ResearchDetailPages. I have marked them in the pseudocode below with a comment that starts off with ` ***** #NOTE:`.


```python
    def test_filter_multiple_author_profiles(self):
        start_time = time.time()
        detail_page_1 = detail_page_factory.ResearchDetailPageFactory(
            parent=self.library_page,
        )
        profile_a = detail_page_1.research_authors.first().author_profile
        detail_page_2 = detail_page_factory.ResearchDetailPageFactory(
            parent=self.library_page,
        )
        profile_b = detail_page_2.research_authors.first().author_profile
        # Make author of first page also an author of the second page
        relations_factory.ResearchAuthorRelationFactory(
            research_detail_page=detail_page_2,
            author_profile=profile_a,
        )

       # ***** NOTE: This line will NOT a populated queryset *****
        daniels_test_queryset_1 = self.library_page._get_research_detail_pages()

        response = self.client.get(
            self.library_page.url,
            data={"author": [profile_a.id, profile_b.id]},
        )

       # ***** NOTE: This line will return a populated queryset *****
        daniels_test_queryset_2 = self.library_page._get_research_detail_pages()

        research_detail_pages = response.context["research_detail_pages"]
        self.assertNotIn(detail_page_1, research_detail_pages)
        self.assertIn(detail_page_2, research_detail_pages)
```

What I am finding is that the first attempt to retrieve the queryset will return a blank queryset. The second attempt, underneath the `.get()` request to the client however, returns 2 ResearchDetailPages as expected.

Also, if I delete the `.get()` request, both of the attempts return blank querysets.

I have tried to save the pages after creating them with the factory, but that makes no difference.

I am also experiencing the same thing in the test `test_pagination`.

Would you have any guess as to why this might be happening?

Here is a link to my [WIP PR](https://github.com/MozillaFoundation/foundation.mozilla.org/pull/10488) if you would like to take a look at the other updates I have so far.

Thanks in advance!

**Comment by tbrlpld** [comment]
@danielfmiranda I wonder if this issue comes up because of the `locale=Locale.get_active()` filtering.

https://github.com/MozillaFoundation/foundation.mozilla.org/blob/main/network-api/networkapi/wagtailpages/pagemodels/research_hub/library_page.py#L182

There may not be an "active" locale because we are not in a request context. You could try using `locale=self.locale` because the library page itself has locale information. 

Maybe it works on the second attempt because of the request in between 🤔 I would try removing that `get` complete and see if the queryset is empty in both cases.

**Comment by danielfmiranda** [comment]
Hi everyone! 

After some discussion with @jhonatan-lopes, we have decided that this ticket should be put on hold until [https://github.com/MozillaFoundation/foundation.mozilla.org/issues/10392](https://github.com/MozillaFoundation/foundation.mozilla.org/issues/10392) is complete. 


This is due to the fact that the PR related to #10392 is going to be making some changes to the functions that we are testing/updating in this ticket.

**Comment by danielfmiranda** [comment]
Hi Everyone, this has been merged into main, however since its just an update of the tests and they have already passed on the PR, I  am going to move this to "ready for prod".

Thanks!
